### ISSUE TITLE
Query-expressions that construct maps and tables should not always have error type 

### ISSUE THREAD

**Issue by SandaruJayawardana** [ballerina-platform/ballerina-lang#36695]
**Description:**
$title. This should be fixed after fixing https://github.com/ballerina-platform/ballerina-lang/issues/36181. 

**Steps to reproduce:**

Ex.
```ballerina
type Employee record {|
    readonly int id;
    string firstName;
|};

public function main() {
    table<Employee> key(id) employees = table [
        {id: 1, firstName: "John"}
    ];

    table<Employee> key(id) highPaidEmployees = table key(id) from var e in employees
                            select e; // expected 'table<Employee> key(id)', found '(table<Employee> key(id)|error)'
// Here, both `from-clause` and `select-clause` will not complete early with an error value. 
// Therefore, for this scenario, compiler should allow this statement.
// Same for map case also
}
```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by pcnfernando** [comment]
Table case which is described in https://github.com/ballerina-platform/ballerina-lang/issues/36686 will be fixed with https://github.com/ballerina-platform/ballerina-lang/issues/36686

**Comment by pcnfernando** [comment]
@SandaruJayawardana shall we back-port the fix to applicable branches

**Comment by SandaruJayawardana** [comment]
> @SandaruJayawardana shall we back-port the fix to applicable branches
Sure. I will send this fix to `2201.2.x`.

**Comment by SandaruJayawardana** [comment]
https://github.com/ballerina-platform/ballerina-lang/pull/37925 will fix this issue in `2201.2.x` branch.
